lib/common/utils.js

```diff
@@ -179,22 +188,22 @@ export const relativeTo = function relativeTo() {
 
 export const flatRoutes = function flatRoutes(router, _path = '', routes = []) {
   router.forEach((r) => {
-    if (!r.path.includes(':') && !r.path.includes('*')) {
-      /* istanbul ignore if */
-      if (r.children) {
-        if (_path === '' && r.path === '/') {
-          routes.push('/')
-        }
-        flatRoutes(r.children, _path + r.path + '/', routes)
-      } else {
-        _path = _path.replace(/^\/+$/, '/')
-        routes.push(
-          (r.path === '' && _path[_path.length - 1] === '/'
-            ? _path.slice(0, -1)
-            : _path) + r.path
-        )
+    if ([':', '*'].some(c => r.path.includes(c))) {
```

Comment:
```js
const NAMINGISHARDex = /\*|:/
// ...
NAMINGISHARDex.test(r.path)
```

Reply:
Won't implement as it hurts readability/maintainability.

Reply:
Testing a well-named regex such as `CONTAINS_SEMICOLON_OR_ASTERISK` isn't worse readability than collection methods.